### ISSUE TITLE
Add random flips

### ISSUE THREAD

**Issue by peterdudfield** [openclimatefix/nowcasting_dataset#78]
When getting batch data from gcp/aws, want option of adding random flips. 
This helps will help reduce over fitting

**Comment by JackKelly** [comment]
Good idea! Let's definitely give this a go, but this might not help as much as it does in most computer vision tasks because orientation matters quite a lot for weather:  in particular, the UK's prevailing wind direction is from the South West.  So flipped images might be physically unrealistic.

Do you think we should save randomly flipped data to disk?  Or should we compute the random flips on-the-fly during training? (the thinking being that flips are computationally very cheap; but could significantly increase the size on disk!)

**Comment by peterdudfield** [comment]
Yea was thinking doing this 'on-the-fly', 

good point, that the prevailing wind direction is from the southwest, so it might not be that useful. Although 
 - I dont think the UKs wind rose is so uni-drirectional that this wont be useful - https://www.southwest-environmental.co.uk/further%20info/in%20depth/wind_data_for_yeolvilton.html
- Might be useful in the long term for other countries where the prevailing wind 

Yea havent really thought about 'size on disk', but might be something we can try and see

**Comment by peterdudfield** [comment]
@JackKelly any thought on what to do with x and y coords when flipping. 

I'm tempted to leave 'sat_x_coords', 'sat_y_coords' and nwp the same, but change the 'pv_yield' coords, and the satellite image will be different

**Comment by JackKelly** [comment]
Now that I think about it some more, flipping will have no effect for fully attentional models (because fully attentional models are set-to-set: The order of the inputs doesn't matter).  So, tbh, I wouldn't worry about flipping for now :)  Sorry, I should've thought about this earlier!

Also, I'm not aware of any evidence that we're overfitting yet.  Have you seen evidence that we're overfitting?  If we're not overfitting, then do we need to augment the training dataset?

TBH, my hunch is that it's more important to spend the effort growing the 'real' dataset before we reach for augmentations.

If we do revisit flipping in the future, I think we need to be 'honest' with the model w.r.t. the geospatial position for each pixel :)

**Comment by peterdudfield** [comment]
Yea I agree a 'real' dataset will provide more information, just I'm to have some sort of randomness. 

https://app.neptune.ai/o/OpenClimateFix/org/predict-pv-yield/e/PRED-245/charts
shows some signs of over fitting - i.e training error sill going dow, validation error is I think not going down anymore. 
p.s. this experiment has been running a long time.

Yea maybe its not as simple as thought, I was thinking of moving the pv coordinate once the image is flipped, but then the emdding of the pv system id might not be quite right. 

Perhaps can leave it for the moment
